lib/core/src/Cardano/Wallet.hs

```diff
@@ -1417,6 +1417,9 @@ quitStakePool ctx wid argGenChange pwd = db & \DBLayer{..} -> do
     unless (isDelegatingTo anyone (walMeta ^. #delegation)) $
         throwE ErrQuitStakePoolNotDelegating
 
+    when (any statusNotDelegating (walMeta ^. #delegation . #next)) $
+        throwE ErrQuitStakePoolNextNotDelegating
```

Comment:
The check should be on the most recent pending delegation only. If i quit delegation an epoch E+1, but I join again at epoch E+2, then the API should accept any delegation certificate to replace the most recent one at E+2.

Also, joining likely needs some adjustments. Right now, we do prevent delegation if the wallet is delegating toa particular pool at any stage. But, I could be in a situation where I have:

E: delegate to P
E+1: quit
E+2: -

It'd be totally okay to delegate to P at this stage but the API would prevent it. Again, when making assertions on the delegation status in at the API, I think we should only be looking at the status at E+2.

Reply:
I have adjusted join and quit stake pool as pointed :point_up_2: and updated integration tests.

af12bb8
Adjust quitStakePool/joinStakePool to not allow joining/quitting only when last pending interferes

d70ae21
Adjust integration tests...

I think that integration tests are fairly exhaustive (`STAKE_POOLS_NEXT`), but they are quite long:

```
Finished in 566.3162 seconds
12 examples, 0 failures
```